src/Nethermind/Nethermind.Blockchain/Processing/BlockProcessor.cs

```diff
@@ -239,10 +239,15 @@ void ProcessTransaction(Transaction currentTx, int index)
                 foreach (Transaction currentTx in transactions)
                 {
                     // No more gas available in block
-                    if (currentTx.GasLimit > block.Header.GasLimit - block.GasUsed)
+                    long gasRemaining = block.Header.GasLimit - block.GasUsed;
```

Comment:
cannot change consensus inside processor

Reply:
This is special path for producing blocks, we can extract it to either derived class or a strategy inside block processor. But its fine.